app/services/csv_upload_normalizer.rb

```diff
@@ -11,10 +11,14 @@ def initialize(path, druid_headers: %w[Druid druid])
     @druid_headers = druid_headers
   end
 
-  # Reads and normalizes CSV.
+  # Reads uploaded file (could be csv or Excel) and normalizes to CSV.
   # This includes handling BOM and druids without prefixes.
   def read
-    table = CSV.open(path, 'rb:bom|utf-8', headers: true).read
+    raise 'Unsupported upload file type' unless %w[.csv .ods .xls .xlsx].include? File.extname(path)
+
+    spreadsheet = Roo::Spreadsheet.open(path, { csv_options: { encoding: 'bom|utf-8' } }) # open the spreadsheet
```

Comment:
are the encoding options still necessary? Does roo take care of this automatically?

Reply:
I believe that it does for Excel files, but if you start with a CSV it may not handle automatically.  The roo documentation lists this option for input encoded CSV files, so it seems reasonable to include it. We could verify on stage by deploying and seeing (if Andrew has different encoded files).

Reply:
You can create a CSV with a BOM by saving in Excel as "CSV UTF-8" (the default CSV option in Mac Excel).

Reply:
Actually I just noticed there is an existing test for regular vs UTF-8 CSVs (with both kinds in the fixture files), see https://github.com/sul-dlss/argo/pull/3446/files#diff-1b4024a38dec55440d3468e90ef2fa1be8ba1b1bd4a11d31059f481efc496cfbL14-L27

Both tests pass in this PR, but the UTF-8 CSV test fails in this PR if I remove the encoding options from the `Roo::Spreadsheet.open` on line 19.  So at least the tests suggest this encoding option is needed and is working as expected.